lib/index.d.ts

```diff
@@ -39,3 +45,78 @@ export class StateController<State> {
   /** @internal Unsubscribe the listener callback that was passed to subscribeUpdates */
   unsubscribeUpdates(listener: (props: Partial<State>) => void): void;
 }
+
+declare namespace Component {
+    export interface ConfigOptions<State> {
+        /** Function transforming state object to virtual dom tree */
+        template(state: State): VNode;
```

Comment:
This is official jsdoc. We also need to include `hooks` and `appState` in config.

```
   * @property {function} template - function transforming state object to virtual dom tree
   * @property {object} [helpers={}] - properties and functions injected automatically into template state object
   * @property {object} [routes={}] - object mapping string route expressions to handler functions
   * @property {object} [appState={}] - (app root component only) state object to share with nested descendant components;
   * if not set, root component shares entire state object with all descendants
   * @property {object} [defaultState={}] - default entries for component state
   * @property {object} [hooks={}] - extra rendering/lifecycle callbacks
   * @property {function} [hooks.preUpdate] - called before an update is applied
   * @property {function} [hooks.postUpdate] - called after an update is applied
   * @property {boolean} [updateSync=false] - whether to apply updates to DOM
   * immediately, instead of batching to one update per frame
   * @property {boolean} [useShadowDom=false] - whether to use Shadow DOM
   * @property {string} [css=''] - component-specific Shadow DOM stylesheet
   * @example
```